src/feedspora/feedspora_runner.py

```diff
@@ -330,6 +292,108 @@ def test_output(self, text):
 
         return self.output_test(output)
 
+    # pylint: disable=too-many-branches
+    def shorten_url(self, the_url):
+        '''
+        Apply configured URL shortener (if present) to the provided link and
+        return the result.  If anything goes awry, return the unmodified link.
+        '''
+        to_return = the_url
+        # Default
+        short_options = {'timeout': 3}
+        short_options.update(self._url_shortener_opts)
+        if the_url and self._url_shortener and (self._url_shortener != 'none'):
+            try:
+                shortener = pyshorteners.Shortener(**short_options)
+                if self._url_shortener == 'adfly':
+                    to_return = shortener.adfly.short(the_url)
+                elif self._url_shortener == 'bitly':
+                    to_return = shortener.bitly.short(the_url)
+                elif self._url_shortener == 'chilpit':
+                    to_return = shortener.chilpit.short(the_url)
+                elif self._url_shortener == 'clkru':
+                    to_return = shortener.clkru.short(the_url)
+                elif self._url_shortener == 'dagd':
+                    to_return = shortener.dagd.short(the_url)
+                elif self._url_shortener == 'isgd':
+                    to_return = shortener.isgd.short(the_url)
+                elif self._url_shortener == 'osdb':
+                    to_return = shortener.osdb.short(the_url)
+                elif self._url_shortener == 'owly':
+                    to_return = shortener.owly.short(the_url)
+                elif self._url_shortener == 'post':
+                    to_return = shortener.post.short(the_url)
+                elif self._url_shortener == 'qpsru':
+                    to_return = shortener.qpsru.short(the_url)
+                elif self._url_shortener == 'soogd':
+                    to_return = shortener.soogd.short(the_url)
+                elif self._url_shortener == 'tinycc':
+                    to_return = shortener.tinycc.short(the_url)
+                elif self._url_shortener == 'tinyurl':
+                    to_return = shortener.tinyurl.short(the_url)
+                else:
+                    all_shorteners = ' '.join(shortener.available_shorteners)
+                    logging.error('URL shortener %s is unimplemented!',
+                                  self._url_shortener)
+                    logging.info('Available URL shorteners: %s',
+                                 all_shorteners)
+                # Sanity check!
+                if len(to_return) > len(the_url):
+                    # Not shorter?  You're fired!
```

Comment:
:+1:

Reply:
I tripped over that condition with some misbehaving shorteners that were returning an entire web page explaining why they weren't working!